src/CivGame/Game/Segment_2c84.cs

```diff
@@ -286,19 +286,19 @@ private void F0_2c84_01d8_OrdersMenu(short playerID, short unitID)
 					}
 					else
 					{
-						if (this.oParent.CivState.TerrainMultipliers[terrainID].Multi1 >= 0)
+						if (this.oParent.CivState.TerrainModifications[terrainID].IrrigationEffect >= 0)
 						{
 							// Instruction address 0x2c84:0x0342, size: 5
 							this.oParent.MSCAPI.strcat(0xba06, " Change to ");
 
 							// Instruction address 0x2c84:0x035d, size: 5
 							this.oParent.MSCAPI.strcat(0xba06,
 								this.oParent.CivState.Terrains[this.oCPU.ReadInt16(this.oCPU.DS.Word,
-									(ushort)(0x2ba6 + this.oParent.CivState.TerrainMultipliers[terrainID].Multi1 * 2))].Name);
+									(ushort)(0x2ba6 + this.oParent.CivState.TerrainModifications[terrainID].IrrigationEffect * 2))].Name);
```

Comment:
This can be changed to:
```c#
TerrainModification modification = this.oParent.CivState.TerrainModifications[terrainID];
TerrainDefinition? resultTerrain = Array.Find(this.oParent.CivState.Terrains, terrain => terrain.TerrainChangeID == modification.IrrigationEffect);
Debug.Assert(resultTerrain != null);

this.oParent.MSCAPI.strcat(0xba06, resultTerrain.Name);
```
Same with the mining order below on line 323:
```c#
TerrainModification modification = this.oParent.CivState.TerrainModifications[terrainID];
TerrainDefinition? resultTerrain = Array.Find(this.oParent.CivState.Terrains, terrain => terrain.TerrainChangeID == modification.MiningEffect);
Debug.Assert(resultTerrain != null);

this.oParent.MSCAPI.strcat(0xba06, resultTerrain.Name);
```